--- origsrc/eina-0.9.9.063/src/lib/eina_mempool.c	2009-09-07 08:20:17.000000000 -0500
+++ src/eina-0.9.9.063/src/lib/eina_mempool.c	2010-02-03 00:32:17.576548500 -0600
@@ -169,7 +169,7 @@ eina_mempool_init(void)
    _backends = eina_hash_string_superfast_new(NULL);
 
    /* dynamic backends */
-   _modules = eina_module_list_get(NULL, PACKAGE_LIB_DIR "/eina/mp/", 0, NULL, NULL);
+   _modules = eina_module_list_get(NULL, PACKAGE_LIB_DIR "/eina/mp/" MODULE_ARCH, 0, NULL, NULL);
 
    path = eina_module_environment_path_get("HOME", "/.eina/mp/");
    _modules = eina_module_list_get(_modules, path, 0, NULL, NULL);
--- origsrc/eina-0.9.9.063/src/modules/mp/chained_pool/Makefile.am	2009-09-25 19:20:02.000000000 -0500
+++ src/eina-0.9.9.063/src/modules/mp/chained_pool/Makefile.am	2010-02-03 00:29:58.627601100 -0600
@@ -10,7 +10,7 @@ AM_CPPFLAGS = \
 if EINA_BUILD_CHAINED_POOL
 if !EINA_STATIC_BUILD_CHAINED_POOL
 
-controllerdir = $(libdir)/eina/mp
+controllerdir = $(libdir)/eina/mp/$(MODULE_ARCH)
 controller_LTLIBRARIES = eina_chained_mempool.la
 
 eina_chained_mempool_la_SOURCES = \
--- origsrc/eina-0.9.9.063/src/modules/mp/ememoa_fixed/Makefile.am	2009-09-25 19:20:02.000000000 -0500
+++ src/eina-0.9.9.063/src/modules/mp/ememoa_fixed/Makefile.am	2010-02-03 00:30:02.498822500 -0600
@@ -12,7 +12,7 @@ AM_CPPFLAGS = \
 if EINA_BUILD_EMEMOA_FIXED
 if !EINA_STATIC_BUILD_EMEMOA_FIXED
 
-controllerdir = $(libdir)/eina/mp
+controllerdir = $(libdir)/eina/mp/$(MODULE_ARCH)
 controller_LTLIBRARIES = eina_ememoa_fixed.la
 
 eina_ememoa_fixed_la_SOURCES = \
--- origsrc/eina-0.9.9.063/src/modules/mp/ememoa_unknown/Makefile.am	2009-09-25 19:20:02.000000000 -0500
+++ src/eina-0.9.9.063/src/modules/mp/ememoa_unknown/Makefile.am	2010-02-03 00:30:05.163975000 -0600
@@ -12,7 +12,7 @@ AM_CPPFLAGS = \
 if EINA_BUILD_EMEMOA_UNKNOWN
 if !EINA_STATIC_BUILD_EMEMOA_UNKNOWN
 
-controllerdir = $(libdir)/eina/mp
+controllerdir = $(libdir)/eina/mp/$(MODULE_ARCH)
 controller_LTLIBRARIES = eina_ememoa_unknown.la
 
 eina_ememoa_unknown_la_SOURCES = \
--- origsrc/eina-0.9.9.063/src/modules/mp/fixed_bitmap/Makefile.am	2009-09-25 19:20:02.000000000 -0500
+++ src/eina-0.9.9.063/src/modules/mp/fixed_bitmap/Makefile.am	2010-02-03 00:30:08.092142400 -0600
@@ -11,7 +11,7 @@ AM_CPPFLAGS = \
 if EINA_BUILD_FIXED_BITMAP
 if !EINA_STATIC_BUILD_FIXED_BITMAP
 
-controllerdir = $(libdir)/eina/mp
+controllerdir = $(libdir)/eina/mp/$(MODULE_ARCH)
 controller_LTLIBRARIES = eina_fixed_bitmap.la
 
 eina_fixed_bitmap_la_SOURCES = \
--- origsrc/eina-0.9.9.063/src/modules/mp/pass_through/Makefile.am	2009-09-25 19:20:02.000000000 -0500
+++ src/eina-0.9.9.063/src/modules/mp/pass_through/Makefile.am	2010-02-03 00:30:13.436448100 -0600
@@ -11,7 +11,7 @@ AM_CPPFLAGS = \
 if EINA_BUILD_PASS_THROUGH
 if !EINA_STATIC_BUILD_PASS_THROUGH
 
-controllerdir = $(libdir)/eina/mp
+controllerdir = $(libdir)/eina/mp/$(MODULE_ARCH)
 controller_LTLIBRARIES = eina_pass_through.la
 
 eina_pass_through_la_SOURCES = \
