--- origsrc/eina-0.9.9.063/src/tests/Makefile.am	2009-09-11 20:50:43.000000000 -0500
+++ src/eina-0.9.9.063/src/tests/Makefile.am	2010-02-03 00:39:17.825585400 -0600
@@ -71,14 +71,13 @@ eina_test_tiler.c
 
 eina_suite_LDADD = @CHECK_LIBS@ $(top_builddir)/src/lib/libeina.la
 
-module_dummydir = $(libdir)/eina/test
-module_dummy_LTLIBRARIES = module_dummy.la
+check_LTLIBRARIES = module_dummy.la
 
 module_dummy_la_SOURCES = \
 eina_test_module_dummy.c
 
 module_dummy_la_LIBADD       = $(top_builddir)/src/lib/libeina.la @EINA_LIBS@
-module_dummy_la_LDFLAGS      = -no-undefined @lt_enable_auto_import@ -module -avoid-version
+module_dummy_la_LDFLAGS      = -no-undefined @lt_enable_auto_import@ -module -avoid-version -rpath `pwd`
 module_dummy_la_LIBTOOLFLAGS = --tag=disable-static
 
 endif
