--- origsrc/eina-1.0.1/src/tests/Makefile.am	2011-04-19 01:50:23.000000000 -0500
+++ src/eina-1.0.1/src/tests/Makefile.am	2011-11-03 17:57:24.141950400 -0500
@@ -65,8 +65,7 @@ eina_test_quadtree.c
 
 eina_suite_LDADD = @CHECK_LIBS@ $(top_builddir)/src/lib/libeina.la -lm
 
-module_dummydir = $(libdir)/eina/test
-module_dummy_LTLIBRARIES = module_dummy.la
+check_LTLIBRARIES = module_dummy.la
 
 module_dummy_la_SOURCES = \
 eina_test_module_dummy.c
@@ -78,7 +77,7 @@ module_dummy_la_CPPFLAGS = 			\
 -I$(top_builddir)/src/lib 			\
 @EFL_EINA_BUILD@
 module_dummy_la_LIBADD       = $(top_builddir)/src/lib/libeina.la @EINA_LIBS@ -lm
-module_dummy_la_LDFLAGS      = -no-undefined @lt_enable_auto_import@ -module -avoid-version
+module_dummy_la_LDFLAGS      = -no-undefined @lt_enable_auto_import@ -module -avoid-version -rpath `pwd`
 module_dummy_la_LIBTOOLFLAGS = --tag=disable-static
 
 endif
